test_api.py

```diff
@@ -31,56 +21,60 @@ def test_duplicate_headers():
 
     """
 
-    resp = request_raw_url(API_URL)
-    headers = resp.getheaders()
-    assert len(set(headers)) == len(headers), 'Duplicate response headers'
+    # Get the raw list of headers (field values not yet merged)
+    raw_headers = requests.get(API_URL).raw.headers
+
+    # Look for duplicate field values
+    for field_name, field_values in raw_headers.iteritems():
+        for value in field_values:
+            assert field_values.count(value) == 1
 
 
 def test_invalid_duplicate_headers():
-    """ Check there are no conflicting headers
+    """ Check there are no conflicting headers.
 
     Only whitelisted header field names are allowed to appear multiple times
     (with different values) in the response.
 
     """
 
-    # List of header keys allowed multiple times
-    allowed_duplicate_headers = (
+    # List of header field names that are allowed multiple times
+    allowed_duplicate_field_names = (
+
         # The "Vary" conditions may be influenced by different (reverse)
         # proxies in the delivery chain.
         'Vary',
     )
 
-    resp = request_raw_url(API_URL)
-    names = resp.headers.keys()
-    assert all(map(
-        # Check if the header is in the list of allowed headers
-        lambda x: x in allowed_duplicate_headers,
-        # in the list of duplicate headers
-        filter(lambda x: names.count(x) > 1, names)
-    ))
+    # Get the raw list of headers (field values not yet merged)
+    raw_headers = requests.get(API_URL).raw.headers
```

Comment:
As noted above this is wrong in the context of this test. This can't be tested with already processed headers.